tests/foreman/ui/test_ansible.py

```diff
@@ -449,6 +449,69 @@ def test_positive_assign_and_remove_ansible_role_to_host(self, target_sat, funct
                 == 'No roles assigned directly to the host'
             )
 
+    @pytest.mark.tier2
+    def test_positive_assign_and_remove_ansible_role_to_hostgroup(
+        self,
+        target_sat,
+        module_org,
+        module_location,
+    ):
+        """Add and remove functionality for ansible roles in hostgroup
+
+        :id: 5d94a484-92c1-4387-ab92-0649e4c4f907
+
+        :steps:
+            1. Import all roles available by default.
+            2. Assign ansible role while creating the hostgroup
+            3. Assign ansible role after creating the hostgroup
+            4. Remove previously added ansible roles from the hostgroup
+
+        :expectedresults: The Ansible Role is successfully added and removed from the hostgroup
+        """
+        SELECTED_ROLE = [
+            'RedHatInsights.insights-client',
+            'redhat.satellite.hostgroups',
+            'redhat.satellite.compute_profiles',
+        ]
+        name = gen_string('alpha').lower()
+        with target_sat.ui_session() as session:
+            synced_all_role = session.ansibleroles.import_all_roles()
+            total_imported_roles = session.ansibleroles.imported_roles_count
+            # verify all roles are synced
+            assert synced_all_role == total_imported_roles
+
+            session.location.select(module_location.name)
+            session.organization.select(module_org.name)
+            # Assign Ansible role(s) while creating the hostgroup.
+            session.hostgroup.create(
+                {
+                    'host_group.name': name,
+                    'ansible_roles.resources': SELECTED_ROLE[:2],
+                }
+            )
+            result = session.hostgroup.read_role(name, SELECTED_ROLE)
+            # verify ansible role(s) assigned properly while creating a host group.
+            assert SELECTED_ROLE[:2] == result
```

Comment:
We verify result we getting with what we've available here like variables, not vice versa
```suggestion
            # verify ansible role(s) assigned properly while creating a host group.
            assert session.hostgroup.read_role(name, SELECTED_ROLE) == SELECTED_ROLE[:2]
```